code/game/objects/effects/plastic_explosive.dm

```diff
@@ -0,0 +1,47 @@
+/obj/effect/plastic_explosive
+	name = "plastic explosives"
+	desc = "Used to put holes in specific areas without too much extra hole."
+	gender = PLURAL
+	icon = 'icons/obj/assemblies.dmi'
+	icon_state = "plastic-explosive2"
+	anchored = TRUE
+	density = FALSE
+	var/obj/item/plastique/parent
+
+/obj/effect/plastic_explosive/Initialize(var/atom/owner_pos, var/atom/target, var/obj/item/plastique/c4)
+	. = ..()
+	parent = c4
+	parent.effect_overlay = src
+	parent.forceMove(src)
+	name = parent.name
+	desc = parent.desc
+	set_position(get_dir(src, target))
+
+/obj/effect/plastic_explosive/proc/set_position(var/dir)
+	var/dir_text = uppertext(dir2text(dir))
+	// DIR = list(pixel_x, pixel_y)
+	var/list/dir_to_pixel = list(
+		"NORTH" = list(0, 32),
+		"NORTHEAST" = list(32, 32),
+		"EAST" = list(32, 0),
+		"SOUTHEAST" = list(32, -32),
+		"SOUTH" = list(0, -32),
+		"SOUTHWEST" = list(-32, -32),
+		"WEST" = list(-32, 0),
+		"NORTHWEST" = list(-32, 32)
+	)
+	var/list/pixel_shifts = dir_to_pixel[dir_text]
+	if(islist(pixel_shifts))
+		pixel_x = pixel_shifts[1]
+		pixel_y = pixel_shifts[2]
+
+/obj/effect/plastic_explosive/examine(mob/user)
+	. = ..()
+	if(Adjacent(user))
```

Comment:
Examine can already handle this, if you pass it a distance argument.

e.g. 

. = ..(distance = 0)
if(.)
   to_chat(whatever)

Reply:
![image](https://user-images.githubusercontent.com/22774890/91462340-5ce71380-e88a-11ea-8742-9311254bda49.png)

Reply:
![image](https://user-images.githubusercontent.com/22774890/91462394-6a040280-e88a-11ea-9c21-ada1e3fcfae8.png)

Reply:
You'll need to add it to the possible args for /obj/examine too then.

Reply:
this distance thing doesn't account for things like adjacent does, so i won't be using it.